src/components/calcite-tile-select-group/calcite-tile-select-group.tsx

```diff
@@ -15,6 +28,30 @@ export class CalciteTileSelectGroup {
   /** Tiles by default move horizontally, stacking with each row, vertical allows single-column layouts */
   @Prop({ reflect: true }) layout?: TileSelectGroupLayout = "horizontal";
 
+  /**
+   * Fired when tiles are selected/deselected
+   * all checked tiles are passed in the event detail
+   */
+  @Event() calciteTileSelectGroupChange: EventEmitter<HTMLCalciteTileSelectElement[]>;
+
+  @Listen("calciteRadioButtonChange")
+  radioButtonChangeHandler(event: CustomEvent): void {
+    this.emitChange(event);
+  }
+
+  @Listen("calciteCheckboxChange")
+  checkboxChangeHandler(event: CustomEvent): void {
+    this.emitChange(event);
+  }
+
+  private emitChange(event: CustomEvent): void {
+    event.stopPropagation();
+    const selected = nodeListToArray(this.el.querySelectorAll("calcite-tile-select")).filter(
+      (el: HTMLCalciteTileSelectElement) => el.checked
+    );
+    this.calciteTileSelectGroupChange.emit(selected);
```

Comment:
In the case of radio-buttons, it would only need to emit the newly checked single value as opposed to an array.  Do you think we should accommodate that so this new custom event only emits the single radio button's information?

Reply:
For future-proofing, I think the custom event detail should be an object with a prop for the selected items. I just checked and we're not consistent in this matter, so I guess we should discuss this separately.

Reply:
WRT Erik's comment, what would be the most useful approach for most use cases? 

1. a convenience prop reflecting the active selection (a la HTMLSelectElement#selectedOptions)
2. a mapping of current selection in the event's payload (a la HTMLCalcite(Pick|Value)ListElement)
3. a mix of 1 & 2 where the event payload has the most recently selected item, which can't be easily determined via a prop
4. no event detail and have users query the DOM for selected items? 👎🏼

Reply:
To move forward and avoid breaking changes, would it be acceptable to not doc the event detail until we decide on the above?

Reply:
@jcfranco Having trouble grokking what you mean by # 2 above.  Could you elaborate?

Reply:
I would lean towards # 1 because that is the pattern in other components.

Reply:
I know I'm in the minority, but I like useful event details (2)  because they are really nice to use in react/preact. You don't have to store a `ref` to the element and then programmatically get the prop storing the data, it just comes over the wire. 

Assuming the `target` is going to be a particular element is not always safe because [events in shadow DOM are pretty complicated](https://pm.dartus.fr/blog/a-complete-guide-on-shadow-dom-and-event-propagation/). I've run in to several situations where I'm making a component using shadow dom and the target becomes my shadow dom component rather than the actual emitting component.

Reply:
I like event details too, I think both event detail and setting the value on the element reference should be supported.  The only question in my mind is which method we should encourage people to use in our documentation.... that part I haven't quite made up my mind about.

Reply:
I'm not a fan of the event details containing duplicate information. It's inconsistent with how native HTML elements work with events and can cause double the breaking changes if we decide to rename a prop. It's much cleaner with once source of information (truth).

Reply:
The thing is, when you use shadow dom in your app, you don't have access to the actual event target so it's not really "duplicating information". Here is a small stencil example with two components: https://github.com/paulcpederson/stencil-events-example

The only way for me to get the actual event target is to use something like `composedPath()[0]` and hope that's correct, which does not work in ie11, and also does not work in legacy edge. Because of that, what you end up doing is saving a ref to the target:


```ts
private selectMenu: HTMLCalciteSelectMenu;

render() {
  <calcite-select 
    ref={(el)=> this.selectMenu = el} 
    onCalciteSelectChange={(e) => { 
      const value = this.selectMenu.selectedOption?.value;
      // do things with value
    }}
  >
    <calcite-option value="1">1</calcite-option>
    <calcite-option value="1">1</calcite-option>
  </calcite-select>
}
```

This is fine, but when you have useful event details it's much easier:

```ts
render() {
  <calcite-select 
    onCalciteSelectChange={(e) => { 
      // do things with e.detail
    }}
  >
    <calcite-option value="1">1</calcite-option>
    <calcite-option value="1">1</calcite-option>
  </calcite-select>
}
```

Why would we make somebody add refs or reach into `composedPath` to get information from an event when `CustomEvents` allow you to just pass useful info in the event itself? `detail` to me is a perfect description of that data ie. "here's useful details about the event". 

That's my main reason I like the event detail. It's always available. In the UI I'm building I've had to ref looooots of times and it just feels like cruft.

Reply:
I think we can make an exception for when the target is in shadow DOM that it can be added to the event.detail. If it's not accessible in light DOM then it's reasonable to have the information in the event.detail.

In many cases, those 'calcite-item's would be slotted though.

As a side note, we aren't really supporting IE11 or LegacyEdge anymore right?

Reply:
> I think we can make an exception for when the target is in shadow DOM

All of our components can be in shadow dom when you are building an application using shadow dom... The constraint is not related to shadow usage in our application, but in the consuming application. The code above is app code, not CC code.

Reply:
@paulcpederson in your example case is the problem the targets when `my-component` is used in an app or another component? I'm having trouble seeing the issue as I can get to either component through the existing event handler.

Reply:
think of `my-component` as an application using shadow dom. When I use `my-nested-component`, I need to save a reference to the `my-nested-component` element. The `target` and `srcElement` will always be `my-component` (my app) because it uses shadow dom. Unless I compose the path and reference an element inside it the event does not include `my-nested-component`. Unless I'm doing something wrong?

So in the UI I'm building, we're using shadow dom, so the event target is never `<calcite-select>`. I have to save references to it.

Reply:
I can't reproduce that. I've updated your sample to use `calcite-select` and the following works:

```ts
// my-component.tsx

<div>
  Click to trigger nested event:
  <calcite-select label="test"
    onCalciteSelectChange={(e) => {
      console.log((e.currentTarget as any).selectedOption.value);
    }}
  >
    <calcite-option>1</calcite-option>
    <calcite-option>2</calcite-option>
  </calcite-select>
</div>
```

Is there something I'm missing?

Reply:
hmmm, you are right it's there when accessing the property, but when you log the event it's null?

<img width="692" alt="Screen Shot 2021-04-28 at 11 10 09 AM" src="https://user-images.githubusercontent.com/1031758/116452288-6a3a1080-a812-11eb-89d8-78dd60222454.png">

Reply:
I mean, if it works it works. I'm weirded out it's not in the actual event log, but I just tried it in the other UI we're building and it works fine? IDK there is probably some event propagation magic messing with the currentTarget prop by the time it gets logged or something.  

So assuming that is a thing you can always do, in this pr I should just add an unsettable prop called `selectedTiles`?